packages/ci/src/lib/schemas.unit.test.ts

```diff
@@ -0,0 +1,108 @@
+import { describe, expect, it } from 'vitest';
+import { ZodError } from 'zod';
+import type { ConfigPatterns } from './models.js';
+import { configPatternsSchema, interpolatedSlugSchema } from './schemas.js';
+
+describe('interpolatedSlugSchema', () => {
+  it('should accept a valid slug', () => {
+    expect(interpolatedSlugSchema.parse('valid-slug')).toBe('valid-slug');
+  });
+
+  it('should accept a slug with {projectName} interpolation', () => {
+    expect(interpolatedSlugSchema.parse('{projectName}-slug')).toBe(
+      '{projectName}-slug',
+    );
+  });
+
+  it('should reject an invalid slug that cannot be fixed by interpolation', () => {
+    expect(() => interpolatedSlugSchema.parse('Invalid Slug!')).toThrow(
+      ZodError,
+    );
+  });
+
+  it('should reject a non-string value', () => {
+    expect(() => interpolatedSlugSchema.parse(123)).toThrow(ZodError);
```

Comment:
Could those 2 live in a each block. I understand the description is lost, just an idea.

Reply:
I would keep them separate, the slug which cannot be fixed by interpolation is a special case. The number is more of a general sanity check.